src/core/util/logger.js

```diff
@@ -13,34 +13,36 @@ const jsonata = require('jsonata');
 class Logger {
     constructor(name = 'default') {
         this.name = name;
+        this.logFunction = null;
         this.reset();
     }
 
-    verbose(message, ctx) {
-        this.__log(message, Logger.__LOG_LEVEL.VERBOSE, ctx);
+    verbose(message, ctx, now) {
```

Comment:
I think it has been introduced to provide tests for that but from the perspective of a feature it makes not much sense for me (if it is used it doesn't improve anything and it opens the possibility to create logs "out of sync")

We could think about having a "timestamp for the related event" .. e.g. if an error is logged with a delay and we want to show the timestamp of the exact event (but this could of course be simply part of the message).

Anyway, can we hide this or something like a "logger.testable"-extension?

Reply:
You can actually put a logging facade in front of the logger, which is used by the user and still have the actual logger implementation testable by exposing the current timestamp.

Reply:
So, it would be something for a later step. Would be ok for me